### PR TITLE
chore: Add package.json integrity tools

### PR DESCRIPTION
**Pull request by zetlen**
## This PR is a:

- [ ] New feature
- [x] Enhancement/Optimization
- [ ] Refactor
- [ ] Bugfix
- [x] Test for existing code
- [ ] Documentation

## Summary

- Update [@magento/eslint](https://github.com/magento-research/magento-eslint)
  to `1.3.0` to include
  [eslint-plugin-package-json](https://github.com/zetlen/eslint-plugin-package-json)
- Modify package scripts to lint package.json files
- Run eslint fixes on all package.json files
- Add `.npmrc` to make default `npm install <dep>` behavior use tilde
prefix

Note: The `.npmrc` should be modified once we're out of the release
candidate cycle.

### CODE REVIEW

File: packages/pwa-buildpack/package.json [hunk 1]
```diff
@@ -52,12 +49,15 @@
...
+  },
   "peerDependencies": {
     "babel-core": "^6.26.0",
     "babel-loader": "^7"
   },
-  "devDependencies": {
-    "babel-cli": "~6.26.0",
-    "rimraf": "~2.6.2"
+  "engines": {
+    "node": ">=8.0.0"
```
Comment (jimbo) [inline packages/pwa-buildpack/package.json]: This should probably be updated to match our root config, [`"10.14.1"`](https://github.com/magento-research/pwa-studio/blob/release/2.0/package.json#L12).

File: packages/upward-js/package.json [hunk 2]
```diff
@@ -55,5 +52,8 @@
   },
   "peerDependencies": {
     "express": "^4.16.3"
+  },
+  "engines": {
+    "node": ">=8.0.0"
```
Comment (jimbo) [inline packages/upward-js/package.json]: Likewise, probably `"10.14.1"`.

File: packages/pwa-buildpack/package.json (hunk 1 above)
Comment (zetlen) [reply packages/pwa-buildpack/package.json]: Hey, you're right! Not technically part of this PR, but we might as well seal in the change.

```suggestion
    "node": ">=10.14.1"
```

File: packages/upward-js/package.json (hunk 2 above)
Comment (zetlen) [reply packages/upward-js/package.json]:
```suggestion
    "node": ">=10.14.1"
```

**Comment by zetlen** [comment]
@jimbo I believe we're doing two-space everywhere else--otherwise this changeset would be much larger. The four-space package.jsons were in the minority.

**Comment by zetlen** [comment]
CI just passed, but [today's CircleCI service degradation](https://github.com/magento-research/pwa-studio/pull/737) means the GitHub checks aren't updating. @magento-research/core-pwa-devs Since this PR might cause ongoing merge conflicts, I want to get it merged today and fanned out to the rest of the open PRs. Would you mind re-reviewing?

File: package.json [hunk 3]
```diff
@@ -42,11 +42,13 @@
...
   },
   "devDependencies": {
+    "@babel/core": "~7.2.2",
     "@magento/directive-parser": "~0.1.1",
-    "@magento/eslint-config": "~1.2.3",
+    "@magento/eslint-config": "~1.3.0",
     "@storybook/addon-actions": "~3.4.2",
     "@storybook/addons": "~3.4.6",
     "@storybook/react": "~4.0.11",
+    "acorn": "~6.0.5",
```
Comment (sirugh) [inline package.json]: What is using acorn? I couldn't find any existing code or changes in this PR that indicate its use.

Comment (zetlen) [reply package.json]: It's a peer dependency of a couple of eslint plugins. NPM was complaining.

File: packages/upward-spec/package.json [hunk 4]
```diff
@@ -1,37 +1,37 @@
...
+    "graphql": "~0.13.2",
+    "graphql-tag": "~2.10.0",
+    "js-yaml": "~3.12.0",
+    "node-fetch": "~2.2.1",
+    "tap-diff": "~0.1.1",
+    "tap-xunit": "~2.3.0",
+    "tape": "~4.9.1"
+  },
+  "engines": {
+    "node": ">=8.0.0"
```
Comment (sharkySharks) [inline packages/upward-spec/package.json]: bump?

Comment (zetlen) [reply packages/upward-spec/package.json]: Thanks, I will.

```suggestion
    "node": ">=10.14.1"
```

**Comment by zetlen** [comment]
@sharkySharks Almost no dependencies have changed; the new lint rule [package-json/order-properties](https://github.com/zetlen/eslint-plugin-package-json/blob/master/docs/rules/order-properties.md) has just reordered the properties in package.json and the diff is hard to read.

**Pull request merged**